node/src/main/scala/node/comm/GrpcClient.scala

```diff
@@ -0,0 +1,62 @@
+package node.comm
+
+import cats.effect.{Async, Resource, Sync}
+import cats.syntax.all.*
+import io.grpc.*
+import io.grpc.netty.NettyChannelBuilder
+
+import scala.concurrent.Promise
+
+class GrpcClient[F[_]: Async] private (private val channel: ManagedChannel) {
+  def send[Req, Resp](msg: Req)(implicit method: MethodDescriptor[Req, Resp], logger: Logger): F[Resp] = for {
+    response <- Async[F].fromFuture {
```

Comment:
We have syntax `.asEffect` for Future

Reply:
On another hand you cannot just call it on future since you need to put Sync[F].delay here. So this is good.